dvc/utils/hydra.py

```diff
@@ -25,20 +26,31 @@ def compose_and_dump(
         output_file: File where the composed config will be dumped.
         config_dir: Folder containing the Hydra config files.
             Must be absolute file system path.
+        config_module: Module containing the Hydra config files.
+            Ignored if `config_dir` is not `None`.
         config_name: Name of the config file containing defaults,
             without the .yaml extension.
         overrides: List of `Hydra Override`_ patterns.
 
     .. _Hydra Override:
         https://hydra.cc/docs/advanced/override_grammar/basic/
     """
-    from hydra import compose, initialize_config_dir
+    from hydra import compose, initialize_config_dir, initialize_config_module
     from omegaconf import OmegaConf
 
     from .serialize import DUMPERS
 
-    with initialize_config_dir(config_dir, version_base=None):
-        cfg = compose(config_name=config_name, overrides=overrides)
+    if config_dir is None and config_module is not None:
+        with initialize_config_module(config_module, version_base=None):
+            cfg = compose(config_name=config_name, overrides=overrides)
+    elif config_dir is not None and config_module is None:
+        with initialize_config_dir(config_dir, version_base=None):
+            cfg = compose(config_name=config_name, overrides=overrides)
+    else:
+        raise ValueError(
+            f"Exactly one of {config_dir = } and {config_module = }"
+            "should be different from `None`."
+        )
```

Comment:
Some nitpick formatting (not even sure if it is cleaner or not):

```suggestion
    config_source = config_dir or config_module
    if not config_source:
        raise ValueError("Either `config_dir` or `config_module` should be provided.")
    init = initialize_config_dir if config_dir else initialize_config_module

    with init(config_source, version_base=None):
        cfg = compose(config_name=config_name, overrides=overrides)
```

The "important" part is that I think makes sense to do as stated in the docstring and prefer `config_dir`, for simplicity.
Not really important because DVC would only reach this point after going through the config validation.